### ISSUE TITLE
Editor - code zoom in on context - Super Enhancement 

### ISSUE THREAD

**Issue by Phuket2** [omz/Pythonista-Issues#206]
Sorry, words escaped me how to word the enhancement in the title.  I really don't use any other editors anymore, so not sure if what I suggest below is widely implemented on desktops.

Anyway, my idea for the enhancement is.... via some mechanism ( maybe a selection in the popup menu that appears after a long press) so the editor is only displaying a single class or function from a file.  In this sort of focused view, everything should just work as normal, editing, running etc.  so you are just editing a portion/frame of the file. When you run in that mode the whole file runs as normal.

Sometimes you have a very long script with many functions and classes. And for whatever reason you want them in one file rather than importing them from smaller files.  So now you want to tweak one function or class.  Then you edit/run/ want to edit more. The script does bounce around between runs depending on what you are doing, or you you accidentally press the wrong place. 

Anyway, I feel this would be a huge help to me at least. I find it hard to believe that this would not help of lot of others also.  

Actually, I am very passionate about this. But I have a feeling it's not going to be so well received 😬😱

Btw, I understand, this basically this is what good outliner does. Hoist topics.  And the idea being the same, to let you focus on one single topic at a time.  

Fingers crossed one this one 💅🏽😱

**Comment by controversial** [comment]
Code folding would accomplish this nicely. Perhaps we could have a folding mechanism that integrated with the outline view? 🤔

**Comment by jsbain** [comment]
As an editor action, you can enable the Fold context menu item:
``` 
import editor
_ev=editor._get_editor_tab().editorView()
_ev.allowsFoldingArbitraryRanges=True
_ev.foldingEnabled=True
``` 
This would accomplish Phuket's goal, sort of, but you have to manually fold each time, and once you unfold there is not an easy way to fold again other than by selecting.


It would be possible to write an editor action that found the current class that you are in, get the extent, then use
``` 
_ev.foldRange_animated_userInfo_(NSRange(start,end),True,{'label':'somelabelname'})
``` 
to hide all the non-desired sections, perhaps using the label to compress things at a top level...  actually, someone could probably write an "outline mode" which either folded based on keywords ( class, def, etc) or else used comments to define folding regions in sort of a markdown style headings.

I have no idea whether enabling folding in this way has some side effect which could be bad...

**Comment by controversial** [comment]
Wait, is folding implemented but not exposed?

**Comment by jsbain** [comment]
Yes, has been since at least 1.5 or 1.6 beta...  I think this is the same editor as Editorial, so it supports folding, however i think the buttons dont show in pythonista.

**Comment by Phuket2** [comment]
Hmmm, wow.  Even traditional code folding would be a great improvement. But I still think it's worth thinking outside the box when it comes to coding on mobile devices.  Code folding makes sense on a large screen/with physical keyboards.  Maybe another way is better smaller screen mobile devices.  Again, maybe it's not. Just seems to me that editing could be better in terms of focusing in on portions of code.  But after saying that, I think @omz has done an amazing job with what we have now.  It's not a kiss ass statement, it really does work exceptionally well.

**Comment by zrzka** [comment]
+1 code folding, probably worth filling as a separate issue, -1 for this idea of zoom
